server/routerlicious/packages/routerlicious-base/src/test/nexus/io.spec.ts

```diff
@@ -402,131 +413,476 @@ describe("Routerlicious", () => {
+											verifyExpectedClientSignals(
+												clients.filter((_, index) => index !== 0),
+												[],
+											);
+										});
+
+										it("nacks invalid targetClientID type", () => {
+											const targetedSignal = {
+												targetClientId: true,
+												content: stringSignalContent,
+											};
+
+											sendAndReturnExpectedSignal(clients[0], [
+												targetedSignal,
+											]);
+
+											checkNack(clients[0], "Invalid signal message");
+										});
+
+										it("should nack signals with invalid client ID", () => {
+											const targetedSignal = {
+												targetClientId: clients[1],
+												content: stringSignalContent,
+											};
+											clients[0].socket.send(
+												"submitSignal",
+												"invalidClientID",
+												[targetedSignal],
+											);
+											checkNack(clients[0], "Nonexistent client");
+										});
+
+										it("nacks missing content field", () => {
+											const targetedSignal = {
+												targetClientId: clients[1].clientId,
+											};
+
+											sendAndReturnExpectedSignal(clients[0], [
+												targetedSignal,
+											]);
+
+											checkNack(clients[0], "Invalid signal message");
+										});
+
+										it("nacks invalid optional signal fields", () => {
+											const targetedSignal = {
+												targetClientId: clients[1].clientId,
+												content: stringSignalContent,
+												clientConnectionNumber: false,
+												referenceSequenceNumber: "invalid",
+											};
+
+											sendAndReturnExpectedSignal(clients[0], [
+												targetedSignal,
+											]);
+
+											checkNack(clients[0], "Invalid signal message");
+										});
+
+										it("nacks signal that is not an array", () => {
+											const targetedSignal = {
+												targetClientId: clients[1].clientId,
+												content: stringSignalContent,
+											};
+
+											clients[0].socket.send(
+												"submitSignal",
+												clients[0].clientId,
+												targetedSignal,
+											);
+											checkNack(clients[0], "Invalid signal message");
+										});
+									});
+								} else if (description === "with v1 and v2 clients") {
+									it("should broadcast signal from a v1 client to all connected clients", () => {
+										const expectedSignal = sendValidAndReturnExpectedSignal(
+											clients[0],
+											[stringSignalContent],
+										);
+
+										verifyExpectedClientSignals(clients, expectedSignal);
+									});
+
+									it("should broadcast signal from a v2 client to all connected clients", () => {
+										const targetedSignal: ISentSignalMessage = {
+											content: stringSignalContent,
+										};
+
+										const expectedSignal = sendAndReturnExpectedSignal(
+											clients[1],
+											[targetedSignal],
+										);
+
+										verifyExpectedClientSignals(clients, expectedSignal);
+									});
```

Comment:
Already covered by the base mix case: "${fromClient} should broadcast batched signals to all connected clients"